pkg/porter/credentials.go

```diff
@@ -2,16 +2,100 @@ package porter
 
 import (
 	"fmt"
+	"os"
+	"sort"
+	"time"
 
 	"github.com/deislabs/porter/pkg/context"
-	"github.com/deislabs/porter/pkg/credentials"
+	"github.com/deislabs/porter/pkg/credentialsgenerator"
 	"github.com/deislabs/porter/pkg/printer"
+
+	dtprinter "github.com/carolynvs/datetime-printer"
+	credentials "github.com/deislabs/cnab-go/credentials"
 	"github.com/pkg/errors"
 	yaml "gopkg.in/yaml.v2"
 )
 
-func (p *Porter) PrintCredentials(opts printer.PrintOptions) error {
-	return nil
+// CredentialsFile represents a CNAB credentials file and corresponding metadata
+type CredentialsFile struct {
+	Name     string
+	Modified time.Time
+}
+
+// CredentialsFileList is a slice of CredentialsFiles
+type CredentialsFileList []CredentialsFile
+
+func (l CredentialsFileList) Len() int {
+	return len(l)
+}
+func (l CredentialsFileList) Swap(i, j int) {
+	l[i], l[j] = l[j], l[i]
+}
+func (l CredentialsFileList) Less(i, j int) bool {
+	return l[i].Modified.Before(l[j].Modified)
+}
+
+// ListCredentials lists credentials using the provided printer.PrintOptions
+func (p *Porter) ListCredentials(opts printer.PrintOptions) error {
+	dir, err := p.Config.GetCredentialsDir()
+	if err != nil {
+		return errors.Wrap(err, "unable to determine credentials directory")
+	}
+
+	// TODO: should this go somewhere else?
+	// i.e., where should logic for init-ing dirs needed by porter go?
+	// e.g., on 'porter creds list' with a fresh porter install/home
+	if ok, _ := p.Context.FileSystem.DirExists(dir); !ok {
```

Comment:
I would prefer that a readonly command `list` not have a side-effect, creating a directory. One way to deal with this a bit easier is to have a function that does the walk and returns a slice of fileinfo's and a error.  Similar to how we have a function that gets the schema and another function that prints the schema. The retrieval isn't tied to the printing.

Then we could return an empty list when the directory doesn't exist.

What do you think?

Reply:
If it's not clear, I'm suggesting that we break this function up into 2 functions. One that traverses the credentials directory that builds up a slice of the credentialsets and another than does the printing. 😀

Reply:
Good call; split into two functions.  Lemme know how this looks when convenient.